calicoctl/resourcemgr/bgppeer.go

```diff
@@ -15,42 +15,46 @@
 package resourcemgr
 
 import (
-	"github.com/projectcalico/libcalico-go/lib/api"
-	"github.com/projectcalico/libcalico-go/lib/api/unversioned"
-	"github.com/projectcalico/libcalico-go/lib/client"
+	"context"
+
+	api "github.com/projectcalico/libcalico-go/lib/apiv2"
+	client "github.com/projectcalico/libcalico-go/lib/clientv2"
+	"github.com/projectcalico/libcalico-go/lib/options"
 )
 
 func init() {
+	ctx := context.Background()
```

Comment:
ugh - sorry i didn't spot this before.

I don't think declaring the context here is very API-friendly.  Really we should be passing context in to each of the functions.

I realize we could just drop this, but I want to pull the resource manager out into libcalico-go and being able to pass in the context may prove important for the AAPIS.  Generally speaking it's important too if (for example) we later want to add timeouts to the CLI commands - which this approach does not easily allow for.

Could you pull out the context and pass it in the various commands down the stack.  It's lower priority than getting the tests running and can be done as a separate PR - but would be useful to have that done before next sprint.